### PR TITLE
Open Solution From Command Line Support

### PR DESCRIPTION
**Pull request by GokulBothe99**
Thank you for your contribution.
Before submitting this PR, please make sure:

- [ ] PR description and commit message should describe the changes done in this PR
- [ ] Verify the PR is pointing to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [ ] Latest Code from master or specific release branch is merged to your branch
- [ ] No unwanted\commented\junk code is included
- [ ] No new warning upon build solution
- [ ] Code Summary\Comments are added to my code which explains what my code is doing
- [ ] Existing unit test cases are passed
- [ ] New Unit tests are added for your development
- [ ] Sanity Tests are successfully executed for New and Existing Functionality
- [ ] Verify that changes are compatible with all relevant browsers and platforms.
- [ ] After creating pull request there should not be any conflicts
- [ ] Resolve all Codacy comments
- [ ] Builds and checks are passed before PR is sent for review
- [ ] Resolve code review comments
- [ ] Update the Help Library document to match any feature changes

## Summary by CodeRabbit

- **New Features**
	- Enhanced command-line interface (CLI) functionality for improved user interaction.
	- Added a new operation to open solutions directly from the CLI.

- **Bug Fixes**
	- Improved error handling for command-line argument parsing.

- **Documentation**
	- Updated method signatures and class visibility for better accessibility and clarity.

### CODE REVIEW

File: Ginger/GingerCoreNET/RunLib/CLILib/DoOptions.cs [hunk 1]
```diff
@@ -28,7 +28,8 @@ public enum DoOperation
         {
             analyze,
             info,
-            clean
+            clean,
+            open
```
[the remark this answers is not in the document]
Comment (GokulBothe99) [reply Ginger/GingerCoreNET/RunLib/CLILib/DoOptions.cs:32]: yes it's added according to purpose

File: Ginger/Ginger/App.xaml.cs [hunk 2]
```diff
@@ -240,47 +242,64 @@
...
+                if (doOptions != null && tempAutoLoad)
+                {
+                    WorkSpace.Instance.UserProfile.AutoLoadLastSolution = false;
+                }
+                StartGingerUI();
+                if (doOptions != null)
+                {
+                    DoOptionsHanlder.Run(doOptions);
+                    WorkSpace.Instance.UserProfile.AutoLoadLastSolution = tempAutoLoad;
+                }
```
[the remark this answers is not in the document]
Comment (GokulBothe99) [reply Ginger/Ginger/App.xaml.cs:292]: yes done

File: Ginger/Ginger/App.xaml.cs [hunk 3]
```diff
@@ -240,47 +242,64 @@
             }
         }
 
-
+        CLIProcessor cLIProcessor;
```
[the remark this answers is not in the document]
Comment (GokulBothe99) [reply Ginger/Ginger/App.xaml.cs:245]: done

File: Ginger/Ginger/App.xaml.cs [hunk 4]
```diff
@@ -302,11 +321,10 @@
             ShowWindow(handle, SW_SHOW);
         }
 
-        private async void RunNewCLI(string[] args)
+        private async void RunNewCLI(ParserResult<object> parserResult)
         {
-            CLIProcessor cLIProcessor = new CLIProcessor();
-            await cLIProcessor.ExecuteArgs(args);
 
+            await cLIProcessor.ProcessResult(parserResult);
```
[the remark this answers is not in the document]
Comment (GokulBothe99) [reply Ginger/Ginger/App.xaml.cs:327]: done

File: Ginger/GingerCoreNET/RunLib/CLILib/CLIProcessor.cs [hunk 5]
```diff
@@ -72,7 +87,7 @@
                 settings.IgnoreUnknownArguments = true;
             });
 
-            int result = await parser.ParseArguments<RunOptions, GridOptions, ConfigFileOptions, DynamicOptions, ScriptOptions, SCMOptions, VersionOptions, ExampleOptions, DoOptions>(args).MapResult(
+            int result = await parserResult.MapResult(
```
[the remark this answers is not in the document]
Comment (GokulBothe99) [reply Ginger/GingerCoreNET/RunLib/CLILib/CLIProcessor.cs:90]: done

File: Ginger/GingerCoreNET/RunLib/CLILib/CLIProcessor.cs [hunk 6]
```diff
@@ -57,12 +57,27 @@ public async Task ExecuteArgs(string[] args)
                 args = newArgs;
             }
 
-            await ParseArgs(args);
+            CLIProcessor cLIProcessor = new CLIProcessor();
+            ParserResult<object> parserResult = args.Length != 0 ? cLIProcessor.ParseArgsOnly(args) : null;
+            await cLIProcessor.ProcessResult(parserResult);
+
```
[the remark this answers is not in the document]
Comment (GokulBothe99) [reply Ginger/GingerCoreNET/RunLib/CLILib/CLIProcessor.cs:63]: done

File: Ginger/GingerCoreNET/RunLib/CLILib/CLIProcessor.cs [hunk 7]
```diff
@@ -57,12 +57,26 @@
...
+        s
+        public ParserResult<object> ParseArgsOnly(string[] args)
+        {
+            var parser = new Parser(settings =>
+            {
+                settings.IgnoreUnknownArguments = true;
+            });
+
+            return parser.ParseArguments<RunOptions, GridOptions, ConfigFileOptions, DynamicOptions, ScriptOptions, SCMOptions, VersionOptions, ExampleOptions, DoOptions>(args);
+
```
[the remark this answers is not in the document]
Comment (GokulBothe99) [reply Ginger/GingerCoreNET/RunLib/CLILib/CLIProcessor.cs:73]: done

File: Ginger/Ginger/App.xaml.cs [hunk 8]
```diff
@@ -240,47 +243,150 @@ public static void OnAutomateBusinessFlowEvent(AutomateEventArgs.eEventType even
...
-                StartGingerUI();// start regular Ginger UI
             }
-            else
+            finally
             {
-                RunNewCLI(e.Args);
+                if (doOptions != null)
+                {
+                    WorkSpace.Instance.UserProfile.AutoLoadLastSolution = tempAutoLoad;
+                }
```
[the remark this answers is not in the document]
Comment (GokulBothe99) [reply Ginger/Ginger/App.xaml.cs:383]: done

File: Ginger/Ginger/App.xaml.cs [hunk 9]
```diff
@@ -240,47 +243,150 @@ public static void OnAutomateBusinessFlowEvent(AutomateEventArgs.eEventType even
...
+
+        /// <summary>
+        /// Extracts the DoOptions object from the parser result if available and the operation is 'open'.
+        /// Otherwise, returns null.
+        /// </summary>
+        /// <param name="parserResult">Parsed command-line arguments.</param>
+        /// <returns>DoOptions object or null.</returns>
+        private DoOptions ExtractDoOptions(ParserResult<object> parserResult)
+        {
+            return parserResult?.Value is DoOptions tempOptions && tempOptions.Operation == DoOptions.DoOperation.open? tempOptions: null;
```
[the remark this answers is not in the document]
Comment (GokulBothe99) [reply Ginger/Ginger/App.xaml.cs:321]: done

File: Ginger/Ginger/App.xaml.cs [hunk 10]
```diff
@@ -302,11 +408,12 @@ public static void ShowConsoleWindow()
...
-        private async void RunNewCLI(string[] args)
+        private async Task RunNewCLI(ParserResult<object> parserResult)
         {
-            CLIProcessor cLIProcessor = new CLIProcessor();
-            await cLIProcessor.ExecuteArgs(args);
-
+            if (parserResult != null)
+            {
+                await cliProcessor.ProcessParsedArguments(parserResult);
+            }
```
[the remark this answers is not in the document]
Comment (GokulBothe99) [reply Ginger/Ginger/App.xaml.cs:416]: done

File: Ginger/Ginger/App.xaml.cs [hunk 11]
```diff
@@ -240,47 +243,150 @@ public static void OnAutomateBusinessFlowEvent(AutomateEventArgs.eEventType even
             }
         }
 
-
+        CLIProcessor cliProcessor;
```
[the remark this answers is not in the document]
Comment (GokulBothe99) [reply Ginger/Ginger/App.xaml.cs:246]: it's used more than one method

File: Ginger/Ginger/App.xaml.cs [hunk 12]
```diff
@@ -240,47 +243,150 @@ public static void OnAutomateBusinessFlowEvent(AutomateEventArgs.eEventType even
...
 
+            InitializeGingerCore();
+
+            if (!WorkSpace.Instance.RunningInExecutionMode)
+            {
+                ProcessGingerUIStartup(doOptions);
+            }
+            else
+            {
+                RunNewCLI(parserResult);
```
[the remark this answers is not in the document]
Comment (GokulBothe99) [reply Ginger/Ginger/App.xaml.cs:278]: done

File: Ginger/Ginger/App.xaml.cs [hunk 13]
```diff
@@ -240,47 +243,158 @@ public static void OnAutomateBusinessFlowEvent(AutomateEventArgs.eEventType even
...
-                StartGingerUI();// start regular Ginger UI
             }
-            else
+            finally
             {
-                RunNewCLI(e.Args);
+                if (doOptions != null)
+                {
+                    WorkSpace.Instance.UserProfile.AutoLoadLastSolution = CheckAutoLoadSolution;
+                }
```
[the remark this answers is not in the document]
Comment (GokulBothe99) [reply Ginger/Ginger/App.xaml.cs:391]: no need

File: Ginger/Ginger/App.xaml.cs [hunk 14]
```diff
@@ -240,47 +243,165 @@ public static void OnAutomateBusinessFlowEvent(AutomateEventArgs.eEventType even
...
+                    ProcessGingerUIStartup(doOptions);
+                }
+                else
+                {
+                    await RunNewCLI(parserResult);
+                }
+            }
+            catch (Exception ex) {
+                Reporter.ToLog(eLogLevel.ERROR, "Unhandled exception in Application_Startup", ex);
+            }
```
[the remark this answers is not in the document]
Comment (GokulBothe99) [reply Ginger/Ginger/App.xaml.cs:285]: it will handle it

**Review by Maheshkale447 — APPROVED** [review]

**Pull request merged**